core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -51,12 +52,12 @@ private[kafka] object LogValidator {
                                                       sourceCodec: CompressionCodec,
                                                       targetCodec: CompressionCodec,
                                                       compactedTopic: Boolean = false,
-                                                      messageFormatVersion: Byte = Record.CURRENT_MAGIC_VALUE,
+                                                      messageFormatVersion: Byte = LogEntry.CURRENT_MAGIC_VALUE,
```

Comment:
I do this validation inside `ProduceRequest`.  If there is more than one entry, we will raise `InvalidRecordException` before the request reaches `LogValidator`.